addons/binding/org.openhab.binding.valloxmv/src/main/java/org/openhab/binding/valloxmv/internal/ValloxMVWebSocket.java

```diff
@@ -0,0 +1,386 @@
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.valloxmv.internal;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.math.BigDecimal;
+import java.net.URI;
+import java.nio.ByteBuffer;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jetty.websocket.api.Session;
+import org.eclipse.jetty.websocket.api.annotations.OnWebSocketClose;
+import org.eclipse.jetty.websocket.api.annotations.OnWebSocketConnect;
+import org.eclipse.jetty.websocket.api.annotations.OnWebSocketError;
+import org.eclipse.jetty.websocket.api.annotations.OnWebSocketMessage;
+import org.eclipse.jetty.websocket.api.annotations.WebSocket;
+import org.eclipse.jetty.websocket.client.ClientUpgradeRequest;
+import org.eclipse.jetty.websocket.client.WebSocketClient;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.library.unit.SIUnits;
+import org.eclipse.smarthome.core.library.unit.SmartHomeUnits;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.types.State;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link ValloxMVWebSocket} is responsible for socket communication with the vallox ventilation unit
+ *
+ * @author Björn Brings - Initial contribution
+ */
+public class ValloxMVWebSocket {
+    private String ip;
+    private ValloxMVHandler voHandler;
+    private WebSocketClient client;
+    private ValloxMVWebSocketListener socket;
+
+    private final Logger logger = LoggerFactory.getLogger(ValloxMVWebSocket.class);
+
+    public ValloxMVWebSocket(ValloxMVHandler voHandler, String ip) {
+        this.voHandler = voHandler;
+        this.ip = ip;
+        client = new WebSocketClient();
+    }
+
+    public void request(ChannelUID channelUID, String updateState) {
+        try {
+            socket = new ValloxMVWebSocketListener(channelUID, updateState);
+            client.start();
+
+            URI destUri = new URI("ws://" + ip + ":80");
+
+            ClientUpgradeRequest request = new ClientUpgradeRequest();
+            logger.debug("Connecting to: {}", destUri);
+            client.connect(socket, destUri, request);
+            socket.awaitClose(2, TimeUnit.SECONDS);
+        } catch (IOException e) {
+            connectionError(e);
+        } catch (Exception e) {
```

Comment:
Do you really need to catch the Exception-class or are there more specific exceptions you could catch like the IOException or maybe only RuntimeException's (which is not that good either). The current code and also catching RuntimeException's might 'hide' programming errors like NullPointerExceptions occurring.

https://www.google.nl/search?q=java%20why%20not%20to%20catch%20exception

Reply:
If you catch all `Exception` you are potentially also catching for example `NullPointerException`s and  they tend to miss a message, so you will be printing the message `null` which will make it troublesome to debug, in this case it is always better to log the complete stacktrace.

Reply:
org.eclipse.jetty.websocket.client.WebSocketClient start/stop throw Exception which I have to handle ...